core/launcher/src/main/java/io/quarkus/launcher/QuarkusLauncher.java

```diff
@@ -33,7 +31,14 @@ public static void launch(String callingClass, String quarkusApplication, Consum
             path = path.substring(0, path.length() - classResource.length());
             URL newResource = new URL(resource.getProtocol(), resource.getHost(), resource.getPort(), path);
 
-            Path appClasses = Paths.get(newResource.toURI());
+            URI uri = newResource.toURI();
+            Path appClasses;
+            if ("jar".equals(uri.getScheme())) {
+                JarURLConnection connection = (JarURLConnection) uri.toURL().openConnection();
```

Comment:
This snippet locks the JAR file in Windows.

Reply:
Call setDefaultUseCaches(false) in the JarlUrlConnection to avoid locking the file in Windows

Reply:
pushed.